dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/ServiceManager.scala

```diff
@@ -0,0 +1,173 @@
+          // calculated are a runtime scoped dependency
+          val buildProjects = session.getProjectDependencyGraph.getUpstreamProjects(project, true).asScala.collect {
+            case runtimeDep if projects.contains(runtimeDep) =>
+              // We need to ensure that we resolve each project that we depend on too
+              facade.resolveProject(runtimeDep, Nil)
+              runtimeDep
+          } :+ project
+
+          val sourceDirsToWatch = projects.flatMap { project =>
+            new File(project.getBuild.getSourceDirectory) ::
+              project.getBuild.getResources.asScala.map(r => new File(r.getDirectory)).toList
+          }
+
+          val watchService = FileWatchService.defaultWatchService(
+            new File(session.getTopLevelProject.getBuild.getDirectory, "target"),
+            200, logger
+          )
+
+          val serviceClassPath = projects.map { project =>
+            new File(project.getBuild.getOutputDirectory)
+          }
+
+          val devSettings = LagomConfig.actorSystemConfig(project.getArtifactId) ++
+            serviceLocatorUrl.map(LagomConfig.ServiceLocatorUrl -> _).toMap ++
+            cassandraPort.fold(Map.empty[String, String]) { port =>
+              LagomConfig.cassandraPort(port) ++ LagomConfig.cassandraKeySpace(cassandraKeyspace)
+            }
+
+          val scalaClassLoader = scalaClassLoaderManager.extractScalaClassLoader(projectDependencies.external)
+
+          val service = Reloader.startDevMode(
+            scalaClassLoader,
+            projectDependencies.external.map(_.getFile),
+            () => {
+              reloadCompile(buildProjects, serviceClassPath)
+            },
+            identity,
+            // todo make configurable
+            sourceDirsToWatch,
+            watchService,
+            new File(project.getBuild.getDirectory),
+            devSettings.toSeq,
+            port
+          )
+
+          // Eagerly reload to start
+          service.reload()
+
+          // Setup trigger to reload when a source file changes
+          service.addChangeListener(() => service.reload())
+
+          LagomKeys.LagomServiceUrl.put(project, service.url())
+
+          runningServices += (project -> service)
+        } catch {
+          case NonFatal(e) =>
+            throw new RuntimeException(s"Failed to start service ${project.getArtifactId}: ${e.getMessage}", e)
+        }
+    }
+  }
+
+  private def reloadCompile(projects: Seq[MavenProject], serviceClassPath: Seq[File]): CompileResult = {
+    try {
+      facade.executeLifecyclePhase(projects, "process-classes")
+      // todo calculate source map
+      CompileSuccess(Map.empty, serviceClassPath)
+    } catch {
+      case NonFatal(e) =>
+        // todo populate properly
+        CompileFailure(new PlayException("Compile failure", "compilation failed", e))
+    }
+
+  }
+
+  def stopService(project: MavenProject) = {
+    runningServices.get(project) match {
```

Comment:
`runningServices` needs to be read either by holding the instance lock, or it should be declared `volatile` to avoid visibility issues.